CMakeLists.txt

```diff
@@ -1,11 +1,31 @@
 # Generated by `boostdep --cmake sort`
 # Copyright 2020 Peter Dimov
+# Copyright 2025 Nigel Stewart
 # Distributed under the Boost Software License, Version 1.0.
 # https://www.boost.org/LICENSE_1_0.txt
 
-cmake_minimum_required(VERSION 3.5...3.16)
+message(STATUS "Using cmake version ${CMAKE_VERSION}")
 
-project(boost_sort VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
+cmake_minimum_required(VERSION 3.10...3.16)
+
+if(BOOST_SUPERPROJECT_VERSION)
+  project(boost_sort VERSION "${BOOST_SUPERPROJECT_VERSION}" LANGUAGES CXX)
+else()
+  project(boost_sort LANGUAGES CXX)
+endif()
+
+# Test coverage in stand-alone mode requires boost dependencies
+
+if(NOT BOOST_SUPERPROJECT_VERSION)
+  if(BUILD_TESTING)
```

Comment:
Maybe rather combine this: `if(BUILD_TESTING AND NOT BOOST_SUPERPROJECT_VERSION)` which shows the intent more clearly, i.e. mirrors your sentence above

And the whole thing should rather go to the test CML where it is required. This makes it easier to see which components are required. E.g. I see only `included_test_exec_monitor` being used